### PR TITLE
Add API for WebsiteContent

### PR DESCRIPTION
**Pull request by noisecapella**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Part of #52

#### What's this PR do?
Adds a website content REST API, a prerequisite to interact with website data 

#### How should this be manually tested?

 - View the list API for website content which should look something like this: `/api/websites/course-id-here/content/`. You should see a smallish list of objects with only a `uuid`, `type`, and `title`. The count should match the number of `WebsiteContent` objects referencing that `Website` with a foreign key. 
 - At the same API, create a new document. You should see a 201 result with the content you specified included in the result. You should see it in the list view as well. A valid `type `is required but you can optionally leave out `title` and `markdown` -- they will be `null` but you can `PATCH` them later. You won't be able to `PATCH` `type` so it needs to be set correctly on create.
 - Go to `/api/websites/course-id-here/content/uuid-for-some-content/` to view the detail view, using one of the uuids from the list view. You should see the same information plus markdown, which was omitted from the list view to reduce data. You should be able to `PATCH` the markdown and title and see the results persist. If you try to patch `type` or `uuid` it should ignore the change.

You should not see any option (at least in this PR) to `DELETE` a `WebsiteContent`.

### CODE REVIEW

**Review by aliceriot — CHANGES_REQUESTED** [review]
test things and they worked, and this all looks pretty straightforward to me. Just had one question really.

File: websites/constants.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 CONTENT_TYPE_PAGE = "page"
 CONTENT_TYPE_FILE = "file"
+CONTENT_TYPES = [CONTENT_TYPE_PAGE, CONTENT_TYPE_FILE]
```
[the remark this answers is not in the document]
Comment (noisecapella) [reply websites/constants.py:5]: Maybe, but that's not being enforced at the database level. The validation is designed to happen within Django Forms which we're not using. https://adamj.eu/tech/2020/01/22/djangos-field-choices-dont-constrain-your-data/

[the remark this answers is not in the document]
Comment (aliceriot) [reply websites/constants.py:5]: oh ok gotcha, I misunderstood how that worked. I was thinking maybe that would automatically then cause DRF to validate for that but I'm not sure. Possibly the [DRF ChoiceField](https://www.django-rest-framework.org/api-guide/fields/#choicefield) could help us out?

[the remark this answers is not in the document]
Comment (noisecapella) [reply websites/constants.py:5]: Good idea, I'll try it out

[the remark this answers is not in the document]
Comment (noisecapella) [reply websites/constants.py:5]: Looks like that `ChoiceField` behavior is done automatically when you set `choices` in the model :man_facepalming:

**Comment by noisecapella** [comment]
FYI I rebased on #85 to get ahead of migration conflicts

**Review by aliceriot — APPROVED** [review]
tested and everything looks good! :+1:

**Pull request merged**